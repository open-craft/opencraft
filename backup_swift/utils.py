# -*- coding: utf-8 -*-
#
# OpenCraft -- tools to aid developing and hosting free software projects
# Copyright (C) 2015-2019 OpenCraft <contact@opencraft.com>
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU Affero General Public License as
# published by the Free Software Foundation, either version 3 of the
# License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU Affero General Public License for more details.
#
# You should have received a copy of the GNU Affero General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.
#
"""
Misc utils for backup_swift.
"""

# Imports #####################################################################

import contextlib
import logging
import requests

# Logging #####################################################################

logger = logging.getLogger(__name__)

# Functions #####################################################################


def ping_heartbeat_url(url):
    """
    Pings heartbeat url.

    :param str url: Url to ping
    :return: True if http request finished and returned proper status, False otherwise.
    """
    try:
        response = requests.get(url, timeout=30)
        return 200 <= response.status_code < 300
    except requests.RequestException:
        return False


@contextlib.contextmanager
def filter_logger(logger_name, new_filter):
    """
    Contextmanager to add a filter to a logger temporarily.
    :param str logger_name: Name of a logger to modify
    :param callable new_filter: Filter to add
    """
    changed_logger = logging.getLogger(logger_name)
    try:
        changed_logger.addFilter(new_filter)
        yield
    finally:
        changed_logger.removeFilter(new_filter)


def filter_swift(log_record):
    """
    Filters log records containing information about the fact that downloaded file was not modified.
    Swift backup spawns quite a log of these, which are not an error condition.
    """
    return '304 Not Modified' not in log_record.getMessage()
