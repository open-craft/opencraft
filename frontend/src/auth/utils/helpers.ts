import { V2Api } from 'global/api';

export const checkAuthAndRefreshToken = async () => {
  const refreshToken = localStorage.getItem('token_refresh');
  let response = null;

  if (refreshToken) {
    try {
      response = await V2Api.authRefreshCreate({
        data: { refresh: refreshToken }
      });
    } catch (e) {
      return false;
    }
  }

  if (response) {
    localStorage.setItem('token_access', response.access);
    return true;
  }

  return false;
};
