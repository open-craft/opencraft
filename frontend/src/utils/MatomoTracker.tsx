import React, { ReactNode } from 'react';
import {
  createInstance,
  MatomoProvider,
  useMatomo
} from '@datapunt/matomo-tracker-react';
import {
  MATOMO_ALIAS_DOMAIN,
  MATOMO_BASE_URL,
  MATOMO_MY_DOMAIN,
  MATOMO_SITE_ID
} from '../global/constants';

interface MatomoTrackerProps {
  children: ReactNode;
}

interface MatomoAdditionalConfiguration {
  [propName: string]: any;
}

interface MatomoConfiguration {
  urlBase: string;
  siteId: number;
  configurations?: MatomoAdditionalConfiguration;
}

interface MatomoUserIdTrackerProps {
  userId: string;
}

/*
 * This wrapper component initializes the Matomo tracking by wrapping the children
 * elements, if the tracking is enabled by providing a valid value for the 'MATOMO_BASE_URL'
 * constant. It allows tracking page views, events and the user id, when available.
 *
 * The 'MATOMO_ALIAS_DOMAIN' constant allows the cross-domain linking of another site with
 * this site so that they are treated as the same site in Matomo.
 */
export const MatomoTracker: React.FC<MatomoTrackerProps> = (
  props: MatomoTrackerProps
) => {
  let element = props.children;
  if (MATOMO_BASE_URL) {
    const matomoConfiguration: MatomoConfiguration = {
      urlBase: MATOMO_BASE_URL,
      siteId: parseInt(MATOMO_SITE_ID, 10)
    };
    if (MATOMO_ALIAS_DOMAIN) {
      const myDomain = `*.${MATOMO_MY_DOMAIN}`;
      const aliasDomain = `*.${MATOMO_ALIAS_DOMAIN}`;
      matomoConfiguration.configurations = {
        enableCrossDomainLinking: true,
        setDomains: [myDomain, aliasDomain]
      };
    }
    element = (
      <MatomoProvider value={createInstance(matomoConfiguration)}>
        {props.children}
      </MatomoProvider>
    );
  }
  return <>{element}</>;
};

/**
 * This is a component passes the provided user id to Matomo if Matomo tracking is enabled.
 * The 'pushInstruction' function exposed by the 'useMatomo()' hook is used to do this.
 *
 * This component is needed as all the components in the registration process are classes and hence
 * the 'yseMatomo' hook cannot be directly used in those components.
 */
export const MatomoUserIdTracker: React.FC<MatomoUserIdTrackerProps> = (
  props: MatomoUserIdTrackerProps
) => {
  const { pushInstruction } = useMatomo();
  if (MATOMO_BASE_URL && props.userId) {
    pushInstruction('setUserId', props.userId);
  }
  return null;
};
