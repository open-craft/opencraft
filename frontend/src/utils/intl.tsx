import React from "react";
import { FormattedMessage, MessageDescriptor } from "react-intl";

type MessageMap = { [k: string]: Omit<MessageDescriptor, 'id'> };

interface Props<T, K extends keyof T> {
    id: K;
    messages: T;
    values?: any;
}


export const WrappedMessage = (props: Props<MessageMap, string>) => {
    const {messages, id, ...values} = props;
    return (
        <FormattedMessage  {...messages[id]} id={id} {...values}/>
    );
};
