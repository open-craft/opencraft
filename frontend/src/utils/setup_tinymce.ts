import 'tinymce/tinymce';
import 'tinymce/themes/silver';
import 'tinymce/skins/ui/oxide/skin.min.css';
import 'tinymce/skins/ui/oxide/content.min.css';
import 'tinymce/plugins/advlist';
import 'tinymce/plugins/autolink';
import 'tinymce/plugins/lists';
import 'tinymce/plugins/link';
import 'tinymce/plugins/image';
import 'tinymce/plugins/charmap';
import 'tinymce/plugins/print';
import 'tinymce/plugins/preview';
import 'tinymce/plugins/anchor';
import 'tinymce/plugins/searchreplace';
import 'tinymce/plugins/visualblocks';
import 'tinymce/plugins/code';
import 'tinymce/plugins/fullscreen';
import 'tinymce/plugins/insertdatetime';
import 'tinymce/plugins/media';
import 'tinymce/plugins/table';
import 'tinymce/plugins/paste';
import 'tinymce/plugins/help';
import 'tinymce/plugins/wordcount';
