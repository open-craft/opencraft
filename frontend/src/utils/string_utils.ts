export const toCamelCase = (s: string) => {
  return s.replace(/([-_][a-z])/gi, ($1) => {
    return $1.toUpperCase().replace('-', '').replace('_', '');
  });
};

export const sanitizeErrorFeedback = (input: {
  [key: string]: Array<string>;
}) => {
  // Loop at each error message and join them.
  // Also convert keys from snake_case to camelCase
  const newObject: any = {};

  Object.keys(input).forEach((key) => {
    newObject[toCamelCase(key)] = input[key].join();
  });

  return newObject;
};

export const capitalizeFirstLetter = (str: string) => {
  return str[0].toUpperCase() + str.slice(1);
};
