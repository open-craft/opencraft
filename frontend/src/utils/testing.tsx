import { createMemoryHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { createRootReducer } from '../global/reducers';

export const setupComponentForTesting = (
  reactContent: JSX.Element,
  storeContents = {}
) => {
  const middleware = applyMiddleware(thunk);

  const store = createStore(
    createRootReducer(createMemoryHistory()),
    storeContents,
    middleware
  );

  return renderer.create(
    <IntlProvider textComponent={React.Fragment} locale="en">
      <Provider store={store}>
        <MemoryRouter>{reactContent}</MemoryRouter>
      </Provider>
    </IntlProvider>
  );
};
