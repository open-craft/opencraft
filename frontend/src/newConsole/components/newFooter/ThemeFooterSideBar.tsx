import * as React from 'react';
import './styles.scss';
import { ConsolePageCustomizationContainer } from 'console/components';
import { ConsolePage } from 'newConsole/components';
import { Col, Row } from 'react-bootstrap';
import { InstancesModel } from 'console/models';
import { connect } from 'react-redux';
import { RootState } from 'global/state';

import { ColorInputField } from 'ui/components/ColorInputField';
import { updateThemeFieldValue } from 'console/actions';
import { WrappedMessage } from 'utils/intl';
import messages from './displayMessages';

interface State {}

interface ActionProps {
  updateThemeFieldValue: Function;
}

interface StateProps extends InstancesModel {}

interface Props extends StateProps, ActionProps {}

export class ThemeFooterSideBarComponent extends React.PureComponent<
  Props,
  State
> {
  private onChangeColor = (fieldName: string, newColor: string) => {
    const instance = this.props.activeInstance;

    if (instance.data) {
      this.props.updateThemeFieldValue(instance.data.id, fieldName, newColor);
    }
  };

  public render() {
    const instance = this.props.activeInstance;
    let themeData;

    if (instance.data && instance.data.draftThemeConfig) {
      themeData = instance.data.draftThemeConfig;
    }

    return (
      <ConsolePage contentLoading={this.props.loading} showSideBarEditComponent>
        {themeData && themeData.version === 1 && (
          <div className="footer-settings">
            <ConsolePageCustomizationContainer>
              <Row>
                <Col>
                  <h2>
                    <WrappedMessage
                      messages={messages}
                      id="newFooterSettings"
                    />
                  </h2>
                </Col>
              </Row>
              <Row>
                <Col>
                  <ColorInputField
                    fieldName="footerBg"
                    initialValue={themeData.footerBg}
                    onChange={this.onChangeColor}
                    messages={messages}
                    loading={instance.loading.includes('draftThemeConfig')}
                  />
                </Col>
              </Row>
              <Row>
                <Col>
                  <ColorInputField
                    fieldName="footerColor"
                    initialValue={themeData.footerColor}
                    onChange={this.onChangeColor}
                    messages={messages}
                    loading={instance.loading.includes('draftThemeConfig')}
                  />
                </Col>
              </Row>
              <Row>
                <Col>
                  <ColorInputField
                    fieldName="footerLinkColor"
                    initialValue={themeData.footerLinkColor}
                    onChange={this.onChangeColor}
                    messages={messages}
                    loading={instance.loading.includes('draftThemeConfig')}
                  />
                </Col>
              </Row>
            </ConsolePageCustomizationContainer>
          </div>
        )}
      </ConsolePage>
    );
  }
}

export const ThemeFooterSideBar = connect<
  StateProps,
  ActionProps,
  {},
  Props,
  RootState
>((state: RootState) => state.console, {
  updateThemeFieldValue
})(ThemeFooterSideBarComponent);
