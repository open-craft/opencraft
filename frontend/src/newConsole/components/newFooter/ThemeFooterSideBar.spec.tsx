import React from 'react';
import { setupComponentForTesting } from "utils/testing";
import { ThemeFooterSideBar } from './ThemeFooterSideBar';

describe('Theme footer sidebar tests', () => {
    it('renders without crashing', () => {
        const tree = setupComponentForTesting(
            <ThemeFooterSideBar />,
            {
                console: {
                    history: {
                        goBack: () => {
                            // empty function
                        }
                    },
                    activeInstance: {
                        loading: [],
                        data: {
                            draftThemeConfig: {
                                version: 1,
                                homePageHeroTitleColor: '#000',
                                homePageHeroSubtitleColor: '#999'
                            }
                        }
                    }
                }
            }
            ).toJSON();
        expect(tree).toMatchSnapshot();
    });
})
