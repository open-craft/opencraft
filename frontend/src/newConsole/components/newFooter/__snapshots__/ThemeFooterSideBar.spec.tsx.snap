// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme footer sidebar tests renders without crashing 1`] = `
<div
  className="console-page"
>
  <div
    className="d-flex justify-content-center align-middle email-activation-alert"
  >
    <i
      className="alert-icon fas fa-exclamation-triangle fa-xs"
    />
    One of your email addresses is pending confirmation. Please confirm it to begin managing your Open edX instance.
  </div>
  <div
    className="new-console-page-container"
  >
    <div
      className="new-console-page-content row"
    >
      <div
        className="pr-4 pl-0 container-fluid"
      >
        <div
          className="justify-content-center m-0 row"
        >
          <div
            className="p-0 m-0 pl-4 col-md-3"
          >
            <div
              className="ml-auto nav"
              onKeyDown={[Function]}
            >
              <span
                className="mr-2"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-angle-left sm"
                />
              </span>
              <span>
                Back
              </span>
            </button>
            <div
              className="footer-settings"
            >
              <div
                className="customization-form"
              >
                <div
                  className="row"
                >
                  <div
                    className="col"
                  >
                    <h2>
                      Footer
                    </h2>
                  </div>
                </div>
                <div
                  className="row"
                >
                  <div
                    className="col"
                  >
                    <div
                      className="color-input form-group"
                    >
                      <label
                        className="form-label"
                      >
                        Footer background color
                      </label>
                      <div
                        className="align-items-center row"
                      >
                        <div
                          className="d-flex align-items-center color-input-field"
                        >
                          <input
                            className="input-field-color form-control"
                            disabled={false}
                            name="footerBg"
                            onClick={[Function]}
                            readOnly={true}
                            value="Not set"
                          />
                          <div
                            className="input-field-preview"
                            style={
                              Object {
                                "backgroundColor": undefined,
                              }
                            }
                          />
                        </div>
                        <div
                          className="info-icon"
                          onBlur={[Function]}
                          onFocus={[Function]}
                          onMouseOut={[Function]}
                          onMouseOver={[Function]}
                        >
                          <i
                            className="fas fa-info-circle"
                          />
                        </div>
                      </div>
                      <p>
                        <button
                          className="reset-value"
                          onClick={[Function]}
                          type="button"
                        >
                          Reset
                        </button>
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  className="row"
                >
                  <div
                    className="col"
                  >
                    <div
                      className="color-input form-group"
                    >
                      <label
                        className="form-label"
                      >
                        Footer text color
                      </label>
                      <div
                        className="align-items-center row"
                      >
                        <div
                          className="d-flex align-items-center color-input-field"
                        >
                          <input
                            className="input-field-color form-control"
                            disabled={false}
                            name="footerColor"
                            onClick={[Function]}
                            readOnly={true}
                            value="Not set"
                          />
                          <div
                            className="input-field-preview"
                            style={
                              Object {
                                "backgroundColor": undefined,
                              }
                            }
                          />
                        </div>
                        <div
                          className="info-icon"
                          onBlur={[Function]}
                          onFocus={[Function]}
                          onMouseOut={[Function]}
                          onMouseOver={[Function]}
                        >
                          <i
                            className="fas fa-info-circle"
                          />
                        </div>
                      </div>
                      <p>
                        <button
                          className="reset-value"
                          onClick={[Function]}
                          type="button"
                        >
                          Reset
                        </button>
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  className="row"
                >
                  <div
                    className="col"
                  >
                    <div
                      className="color-input form-group"
                    >
                      <label
                        className="form-label"
                      >
                        Footer link color
                      </label>
                      <div
                        className="align-items-center row"
                      >
                        <div
                          className="d-flex align-items-center color-input-field"
                        >
                          <input
                            className="input-field-color form-control"
                            disabled={false}
                            name="footerLinkColor"
                            onClick={[Function]}
                            readOnly={true}
                            value="Not set"
                          />
                          <div
                            className="input-field-preview"
                            style={
                              Object {
                                "backgroundColor": undefined,
                              }
                            }
                          />
                        </div>
                        <div
                          className="info-icon"
                          onBlur={[Function]}
                          onFocus={[Function]}
                          onMouseOut={[Function]}
                          onMouseOver={[Function]}
                        >
                          <i
                            className="fas fa-info-circle"
                          />
                        </div>
                      </div>
                      <p>
                        <button
                          className="reset-value"
                          onClick={[Function]}
                          type="button"
                        >
                          Reset
                        </button>
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="pr-0 col-md-9"
          >
            <div
              className="row"
            >
              <div
                className="pr-0 col-md-4"
              >
                <div
                  className="preview-header d-flex justify-content-start"
                >
                  <div
                    className="dropdown-container d-flex flex-column align-items-center"
                  >
                    <a
                      className="nav-link"
                      href="#"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      role="button"
                    >
                      Dashboard
                      <i
                        className="fas xs fa-chevron-down"
                      />
                    </a>
                    <span
                      className="fill-line"
                    />
                    <div
                      className="dropdown-items align-items-center"
                    >
                      <div
                        className="content d-flex flex-column"
                      >
                        <div
                          aria-expanded={null}
                          className="collapse"
                        >
                          <button
                            className="d-inline-block"
                            onClick={[Function]}
                            type="button"
                          >
                            <p>
                              Dashboard
                            </p>
                          </button>
                          <button
                            className="d-inline-block"
                            onClick={[Function]}
                            type="button"
                          >
                            <p>
                              Course Outline
                            </p>
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="pr-0 align-items-start col-md-7"
              >
                <div
                  className="notice-container d-flex justify-content-end"
                >
                  This is only a preview. Publish changes to update the live site
                </div>
              </div>
              <div
                className="preview-box card"
              >
                <div
                  className="container-fluid"
                >
                  <div
                    className="d-flex felx-row main-navigation-menu navigation-menu row"
                    style={
                      Object {
                        "background": undefined,
                      }
                    }
                  >
                    <div
                      className="logo"
                    />
                    <div
                      className="flex-grow-1 d-flex flex-row ml-4 mr-4"
                    >
                      <button
                        className="customizable-link"
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        style={
                          Object {
                            "borderBottomColor": undefined,
                            "color": undefined,
                          }
                        }
                        type="button"
                      >
                        Courses
                      </button>
                      <button
                        className="customizable-link"
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        style={
                          Object {
                            "borderBottomColor": undefined,
                            "color": undefined,
                          }
                        }
                        type="button"
                      >
                        Discover New
                      </button>
                    </div>
                    <div>
                      <div
                        className="navigation-menu navigation-submenu row"
                      >
                        <div
                          className="col"
                        >
                          <button
                            className="customizable-button"
                            onMouseEnter={[Function]}
                            onMouseLeave={[Function]}
                            style={
                              Object {
                                "background": undefined,
                                "borderColor": undefined,
                                "color": undefined,
                              }
                            }
                            type="button"
                          >
                            Register
                          </button>
                          <button
                            className="customizable-button"
                            onMouseEnter={[Function]}
                            onMouseLeave={[Function]}
                            style={
                              Object {
                                "background": undefined,
                                "borderColor": undefined,
                                "color": undefined,
                              }
                            }
                            type="button"
                          >
                            Sign in
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="theme-home"
                >
                  <div
                    className="theme-hero-container row"
                  >
                    <div
                      className="col"
                    >
                      <div
                        className="hero-preview"
                      >
                        <div
                          className="outer-wrapper"
                          style={Object {}}
                        >
                          <div
                            className="title"
                          >
                            <div
                              className="heading-group"
                            >
                              <h1
                                style={
                                  Object {
                                    "color": "#000",
                                  }
                                }
                              >

                              </h1>
                              <p
                                style={
                                  Object {
                                    "color": "#999",
                                  }
                                }
                              >

                              </p>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    className="theme-courses-container row"
                  >
                    <div
                      className="theme-courses-item col-md-3"
                    >
                      <div
                        className="courses-listing-item row"
                        style={
                          Object {
                            "borderBottomColor": undefined,
                          }
                        }
                      >
                        <div
                          className="course-image col"
                        >
                          <img
                            alt="course"
                            src="course_image.jpg"
                          />
                        </div>
                        <div
                          className="organization col"
                          style={
                            Object {
                              "color": undefined,
                            }
                          }
                        >
                          edX
                        </div>
                        <div
                          className="course-number col"
                          style={
                            Object {
                              "color": undefined,
                            }
                          }
                        >
                          DemoX
                        </div>
                        <div
                          className="course-name col"
                        >
                          <button
                            className="customizable-link noHover"
                            onMouseEnter={[Function]}
                            onMouseLeave={[Function]}
                            style={
                              Object {
                                "borderBottomColor": undefined,
                                "color": undefined,
                              }
                            }
                            type="button"
                          >
                            <span>
                              Demonstration Course
                            </span>
                          </button>
                        </div>
                        <div
                          className="course-start col"
                          style={
                            Object {
                              "color": undefined,
                            }
                          }
                        >
                          Starts:
                          Feb 5, 2013
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="theme-footer row"
                >
                  <div
                    className="col"
                  >
                    <div
                      className="custom-footer"
                      style={
                        Object {
                          "background": undefined,
                        }
                      }
                    >
                      <div
                        className="row"
                      >
                        <div
                          className="col-md-4"
                        >
                          <div
                            className="footer-top-links row"
                          >
                            <button
                              className="customizable-link noHover"
                              onMouseEnter={[Function]}
                              onMouseLeave={[Function]}
                              style={
                                Object {
                                  "borderBottomColor": undefined,
                                  "color": undefined,
                                }
                              }
                              type="button"
                            >
                              <span>
                                About
                              </span>
                            </button>
                            <button
                              className="customizable-link noHover"
                              onMouseEnter={[Function]}
                              onMouseLeave={[Function]}
                              style={
                                Object {
                                  "borderBottomColor": undefined,
                                  "color": undefined,
                                }
                              }
                              type="button"
                            >
                              <span>
                                Blog
                              </span>
                            </button>
                            <button
                              className="customizable-link noHover"
                              onMouseEnter={[Function]}
                              onMouseLeave={[Function]}
                              style={
                                Object {
                                  "borderBottomColor": undefined,
                                  "color": undefined,
                                }
                              }
                              type="button"
                            >
                              <span>
                                Contact
                              </span>
                            </button>
                            <button
                              className="customizable-link noHover"
                              onMouseEnter={[Function]}
                              onMouseLeave={[Function]}
                              style={
                                Object {
                                  "borderBottomColor": undefined,
                                  "color": undefined,
                                }
                              }
                              type="button"
                            >
                              <span>
                                Donate
                              </span>
                            </button>
                          </div>
                        </div>
                        <div
                          className="col-md-5"
                        />
                        <div
                          className="openedx-logo col-md-3"
                        >
                          <img
                            alt="Open edX logo"
                            src="openedx-logo-footer.png"
                          />
                        </div>
                      </div>
                      <div
                        className="row"
                      >
                        <div
                          className="logo col-md-1"
                        />
                      </div>
                      <div
                        className="row"
                      >
                        <div
                          className="copyright"
                          style={
                            Object {
                              "color": undefined,
                            }
                          }
                        >
                          © undefined.
                          All rights reserved except where noted. edX, Open edX and their respective logos are registered trademarks of edX Inc.
                        </div>
                      </div>
                      <div
                        className="legal row"
                      >
                        <button
                          className="customizable-link noHover"
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          style={
                            Object {
                              "borderBottomColor": undefined,
                              "color": undefined,
                            }
                          }
                          type="button"
                        >
                          <span>
                            Privacy Policy
                          </span>
                        </button>
                        <button
                          className="customizable-link noHover"
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          style={
                            Object {
                              "borderBottomColor": undefined,
                              "color": undefined,
                            }
                          }
                          type="button"
                        >
                          <span>
                            Terms of Service
                          </span>
                        </button>
                        <button
                          className="customizable-link noHover"
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          style={
                            Object {
                              "borderBottomColor": undefined,
                              "color": undefined,
                            }
                          }
                          type="button"
                        >
                          <span>
                            Honor Code
                          </span>
                        </button>
                        <button
                          className="customizable-link noHover"
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          style={
                            Object {
                              "borderBottomColor": undefined,
                              "color": undefined,
                            }
                          }
                          type="button"
                        >
                          <span>
                            Take free online courses at edX.org
                          </span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
